### PR TITLE
feat(carts): Goodstore presets

### PR DESCRIPTION
**Pull request by lidiyacommercetools**
Added cart presets for the goodstore data. Tests will fail as the product data is not yet finalized.
Carts are mirroring this data: https://docs.google.com/spreadsheets/d/1ehbT8iThVOmS3WdE_3a1eTNlCsSYCVoSOSd1X8hAMDw/edit#gid=1456287537

### CODE REVIEW

File: models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const store = StoreDraft.presets.sampleDataGoodStore
+  .theGoodStore()
+  .build<TStoreDraft>();
+
+const jenniferJones01 = (customerId?: string): TCartDraftBuilder =>
+  CartDraft.presets
+    .empty()
+    .key('jennifer-jones-01')
+    .customerEmail(customer.email)
+    .customerId(customerId)
```
Comment (tylermorrisford) [inline models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.ts:44]: I may be missing something, but should this be `customer.Id`?

File: models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { TCartDraft } from '../../../types';
+import jenniferJones01 from './jennifer-jones-01';
+
+describe('with the preset cart `jenniferJones01`', () => {
+  it('should return a cart draft', () => {
+    const cartDraft = jenniferJones01().build<TCartDraft>();
+
+    expect(cartDraft).toMatchInlineSnapshot();
```
Comment (tylermorrisford) [inline models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.spec.ts:8]: We just need to update (all of) these snapshot tests 💪

File: models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+} from '@commercetools-test-data/shipping-method';
+import { StoreDraft, type TStoreDraft } from '@commercetools-test-data/store';
+import { CartDraft, LineItemDraft } from '../../../../index';
+import { origin } from '../../../constants';
+import type { TCartDraftBuilder } from '../../../types';
+
+const customer = CustomerDraft.presets.sampleDataGoodStore
+  .jenniferJones()
+  .build<TCustomerDraft>();
+const address = AddressDraft.presets.sampleDataGoodStore.jenniferJonesAddress();
```
Comment (tylermorrisford) [inline models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.ts:22]: Question for other reviewers: should we also build the necessary address draft with the build method to ensure that it is type-safe? `AddressDraft.presets.sampleDataGoodStore.jenniferJonesAddress().build<TAddressDraft>();`

**Review by tylermorrisford — APPROVED** [review]
UPDATE: I can see that this work is continued [here](https://github.com/commercetools/test-data/pull/375). LGTM.

File: models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.ts (hunk 3 above)
Comment (jaikamat) [reply models/cart/src/cart/cart-draft/presets/sample-data-goodstore/jennifer-jones-01.ts:22]: This is reasonable to me!

**Review by stephsprinkle — APPROVED** [review]

**Pull request merged**